src/NCronJob/Execution/StartupJobManager.cs

```diff
@@ -8,12 +11,45 @@ internal class StartupJobManager(
     public async Task ProcessStartupJobs(CancellationToken stopToken)
     {
         var startupJobs = jobRegistry.GetAllOneTimeJobs();
-        var startupTasks = startupJobs.Select(definition => CreateExecutionTask(JobRun.Create(observer.Report, definition), stopToken)).ToList();
 
-        if (startupTasks.Count > 0)
+        if (startupJobs.Count == 0)
         {
-            await Task.WhenAll(startupTasks).ConfigureAwait(false);
+            return;
         }
+
+        List<JobRun> jobRuns = [];
+        var startupTasks = startupJobs.Select(definition =>
+        {
+            var jobRun = JobRun.Create(observer.Report, definition);
+            jobRuns.Add(jobRun);
+            return CreateExecutionTask(jobRun, stopToken);
+        });
+
+        await Task.WhenAll(startupTasks).ConfigureAwait(false);
+
+        Exception?[] faults = jobRuns
+            .Where(jr => jr.JobDefinition.ShouldCrashOnStartupFailure == true && jr.CurrentState.Type == JobStateType.Faulted)
+            .Select(jr => jr.CurrentState.Fault)
+            .ToArray();
+
+        if (faults.Length == 0)
```

Comment:
Given line 44: Can `faults` have entries that are just null?
If so, it would make more sense to filter that out on line 33 and continue your logic and remove line 44-47

Reply:
That's something I have to research deeper.
Let me see if I can ensure that every Fault is non null when `JobStateType` is `Faulted`

Reply:
>Let me see if I can ensure that every Fault is non null when JobStateType is Faulted

Fixed. I added a couple of assertions to cover potential future regressions.